### ISSUE TITLE
OOM error with unetr validation when spacing is set at 1.0

### ISSUE THREAD

**Issue by songeeu** [Project-MONAI/research-contributions#26]
**Describe the bug**
OOM at validation for unetr repository when spacing is set at 1mm 

**To Reproduce**
Run the following command for training
python main.py --data_dir=./research-contributions/UNETR/BTCV/dataset/btcv/ --distributed --out_channels=14 --feature_size=16 --batch_size=2 --logdir=benchmark --optim_lr=4e-4 --lrschedule=warmup_cosine --infer_overlap=0.5 --max_epochs=5000 --save_checkpoint --space_x=1.0 --space_y=1.0 --space_z=1.0

**Expected behavior**
If the model fits the vram initially during training, no OOM error during validation 

**Screenshots**
![image](https://user-images.githubusercontent.com/54255954/145692975-43a1cb77-68b2-48a2-bab9-13ae5271aa25.png)

**Environment (please complete the following information):**
 - OS Ubuntu 20.04
 - Python version 3.6.10 
 - MONAI version 0.8.0
 - CUDA/cuDNN version 11.0/8.0.4
 - Also tested on python 3.8.5, MONAI 0.7.0, CUDA/CuDNN 11.2/8.1.0, same issue
 - GPU models and configuration  Tesla V100 16GB DGX-1

**Additional context**
Trying to replicate the experiment mentioned in the unetr paper on BTCV dataset where the isotropic voxel spacing is 1.0 mm.  However, when spacing was set at 1 for both x,y and z, there is always OOM error at the first validation (end of epoch99).

**Comment by ahatamiz** [comment]
Hi @songeeu 
I have tested this issue and runs without any issues on Titan RTX with 24GB VRAM. I recommend changing some of the setting such as setting `batch_size=1` or lowering the `infer_overlap`, although the latter can slightly decrease the final accuracy. 

In addition, although the model weights provided in this repository are not for isotropic spacing, it achieves SOTA benchmarks on the BTCV standard track.

I hope this is useful. 

Thanks

**Comment by songeeu** [comment]
Hi @ahatamiz 

Thank you for taking the time to replicate this issue.  Your repository is amazing and this is great work!

I still have a few questions for you.  What is the cuda/cudnn version that you ran your test on?  I'm still suffering the same memory surge after trying both options so I would love to make sure these are the same. 

For the model weights provided that achieved SOTA on BTCV, were they trained with 1.5mm, 1.5mm and 2mm spacing as set in default or 1mm isometric in the paper?  Are there any performance discrepancies between the two spacing options?

Thank you so much!  Hope you have a great holiday.

**Comment by ahatamiz** [comment]
Hi @songeeu 

Thank you so much for your comments and compliments. 

Regarding the first question, as a background we mainly trained UNETR on DGX servers consisting of V100 32GB GPUs, so we did not encounter the issues with memory. But it is certainly understandable that memory can become a bottleneck using other hardware. For your question, I tested the training on a Titan RTX machine which has a 24GB of VRAM. My CUDA version is 10.2 and I am using PyTorch 1.6.0 for all experiments as well as testing. 

For the second question, the provided weights are for a UNETR model trained with 1.5mm, 1.5mm and 2mm spacing. The reasoning behind it was to relax some of the limitations of memory and make it more accessible. We did observe slight improvement by using isometric resolution, but the gap is very small. 

Best Regards
